cx_Freeze.exception.OptionError: no base named 'console' ('console-cpython-312-x86_64-linux-gnu')

**Desktop:**
 - Platform information : Ubuntu Linux 24.04 (Noble Nimbat)
 - OS architecture : amd64
 - cx_Freeze version : 6.16.0.dev70
 - Python version : 3.12


**Problem/Error**
```
  File "/home/abcd/piaware_builder/package-bookworm/debian/venv/lib/python3.12/site-packages/cx_Freeze/executable.py", line 84, in base
    raise OptionError(f"no base named {name!r} ({name_base!r})")
cx_Freeze.exception.OptionError: no base named 'console' ('console-cpython-312-x86_64-linux-gnu')
make[1]: *** [debian/rules:92: install_mlat-client] Error 1
```

**Additional context**
**Relevant part file debian/rules:**
```
 79 build_mlat-client:
 80         # setup venv for mlat-client build
 81         $(PYTHON3) -m venv --system-site-packages --without-pip $(VENV)
 82         # build cxfreeze for later packaging of mlat-client
 83         $(VENV)/bin/python -m build --skip-dependency-check --no-isolation --wheel -C="--build-option=--plat-name" -C="--build-option=$(PYTHON_HOST_PLATFORM)" --outdir $(CURDIR)/wheels cx_Freeze-6.16.0.dev70
 84         $(VENV)/bin/python -m pip install --no-index --no-deps --ignore-installed --require-virtualenv $(CURDIR)/wheels/cx_Freeze-6.16.0.dev70-*.whl
 85         # build mlat-client proper
 86         $(VENV)/bin/python -m build --skip-dependency-check --no-isolation --wheel -C="--build-option=--plat-name" -C="--build-option=$(PYTHON_HOST_PLATFORM)" --outdir $(CURDIR)/wheels mlat-client 
 87         $(VENV)/bin/python -m pip install --no-index --no-deps --ignore-installed --require-virtualenv $(CURDIR)/wheels/MlatClient-*.whl
 88
 89 install_mlat-client:
 90         # newer cxfreeze deletes --target-dir (see cxfreeze issue #1300), so we have to freeze to a temporary directory and
 91         # install from there
 92         $(VENV)/bin/python $(VENV)/bin/cxfreeze --target-dir=$(CURDIR)/freeze-mlat-client $(VENV)/bin/fa-mlat-client
 93         cp -a $(CURDIR)/freeze-mlat-client/* $(CURDIR)/debian/piaware/usr/lib/piaware/helpers/
 94
```


![image](https://github.com/marcelotduarte/cx_Freeze/assets/28452511/161c691a-5fa3-4bff-849a-760996e1e80a)

--- a/setup.py
+++ b/setup.py
@@ -54,7 +54,7 @@ class BuildBases(setuptools.command.build_ext.build_ext):
         fullname = os.fspath(Path(self.build_lib, filename))
         library_dirs = ext.library_dirs or []
         libraries = self.get_libraries(ext)
-        extra_args = ext.extra_link_args or []
+        extra_args: list = ext.extra_link_args or []
         if PLATFORM.startswith("freebsd"):
             libraries.append("pthread")
         if IS_MINGW or IS_WINDOWS:
@@ -111,6 +111,9 @@ class BuildBases(setuptools.command.build_ext.build_ext):
                 extra_args.extend(get_config_var("BASEMODLIBS").split())
             if get_config_var("LOCALMODLIBS"):
                 extra_args.extend(get_config_var("LOCALMODLIBS").split())
+                # fix for Python 3.12 Ubuntu Linux 24.04 (Noble Nimbat)
+                with contextlib.suppress(ValueError):
+                    extra_args.remove("Modules/_hacl/libHacl_Hash_SHA2.a")
             if IS_MACOS:
                 extra_args.append("-Wl,-export_dynamic")
                 extra_args.append("-Wl,-rpath,@loader_path/lib")
@@ -121,6 +124,7 @@ class BuildBases(setuptools.command.build_ext.build_ext):
                 extra_args.append("-Wl,-rpath,$ORIGIN/../lib")
                 if not self.debug:
                     extra_args.append("-s")
+        link_error = None
         for arg in (None, "-fno-lto", "--no-lto"):
             try:
                 self.compiler.link_executable(
@@ -132,12 +136,17 @@ class BuildBases(setuptools.command.build_ext.build_ext):
                     extra_postargs=extra_args + ([arg] if arg else []),
                     debug=self.debug,
                 )
-            except LinkError:
+            except LinkError as exc:
                 if IS_MINGW or IS_WINDOWS:
                     raise
+                if arg is None:
+                    link_error = exc.args
                 continue
             else:
+                link_error = None
                 break
+            if link_error is not None:
+                raise LinkError from link_error
 
     def get_ext_filename(self, fullname):
         if fullname.endswith("util"):

